src/app/shared/services/save-query.service.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable } from '@angular/core';
+import * as fs from 'fs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SaveQueryService {
+  filePath: string = '';
```

Comment:
if the type is inferred, we usually avoid writing it explicitly
```suggestion
  filePath = '';
```